### PR TITLE
[backport of #20370] New Candidate+Point seeded tracking region producer for Tau HLT tracking

### PR DESCRIPTION
**Pull request by tstreble**
Add a new tracking region producer seeded by candidates + points for Tau HLT Tracking.
To be used for triplet+doublet recovery of HLT paths with regional PF tracking.

This is the 92X back port of #20370

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

File: RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+ *   VerticesSigma:
+ *     origins are defined by vertices from VertexCollection (use maximum MaxNVertices of them)
+ *     z-half-length is defined by nSigmaZVertex * vetex.zError
+ *
+ *   If, while using one of the "Vertices" modes, there's no vertices in an event, we fall back into
+ *   either BeamSpotSigma or BeamSpotFixed mode, depending on the positiveness of nSigmaZBeamSpot.
+ *
+ *   \author Thomas Strebler
+ */
+class  CandidatePointSeededTrackingRegionsProducer : public TrackingRegionProducer
```
Comment (slava77) [inline RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h]: @makortel @VinInn please take a look at this one. 
Thank you.

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h (hunk 1 above)
Comment (makortel) [reply RecoTracker/TkTrackingRegions/plugins/CandidatePointSeededTrackingRegionsProducer.h]: @slava77 I replied in the master PR https://github.com/cms-sw/cmssw/pull/20370#issuecomment-328086074.

**Comment by Martin-Grunewald** [comment]
+1

**Comment by tstreble** [comment]
Postponed due to reimplementation of new module

**Pull request closed without merging**